src/test/java/org/prebid/server/bidder/emx_digital/EmxDigitalBidderTest.java

```diff
@@ -0,0 +1,415 @@
+				assertThat(imp.getBanner()).isEqualTo(expectedBanner);
+			});
+	}
+
+	@Test
+	public void makeHttpRequestsShouldSendRequestToModifiedUrlWithHeaders() {
+		// given
+		final BidRequest bidRequest = BidRequest.builder()
+			.imp(singletonList(Imp.builder()
+				.banner(Banner.builder().w(1).h(1).build())
+				.ext(mapper.valueToTree(ExtPrebid.of(null, ExtImpEmxDigital.of("1", "asd"))))
+				.build()))
+			.device(Device.builder().ip("ip").ua("Agent").language("fr").dnt(1).build())
+			.site(Site.builder().page("myPage").build())
+			.id("request_id")
+			.build();
+
+		// when
+		final Result<List<HttpRequest<BidRequest>>> result = emxDigitalBidder
+			.makeHttpRequests(bidRequest);
+
+		// then
+		assertThat(result.getErrors()).isEmpty();
+		assertThat(result.getValue()).hasSize(1)
+			.first()
+			.satisfies(request ->
+				assertThat(request.getUri()).startsWith("https://test.endpoint.com?t=1000&ts="));
+
+		assertThat(result.getValue()).hasSize(1)
+			.flatExtracting(r -> r.getHeaders().entries())
+			.extracting(Map.Entry::getKey, Map.Entry::getValue)
+			.containsOnly(
+				tuple("Content-Type", "application/json;charset=utf-8"),
+				tuple("Accept", "application/json"),
+				tuple("User-Agent", "Agent"),
+				tuple("X-Forwarded-For", "ip"),
+				tuple("Referer", "myPage"),
+				tuple("DNT", "1"),
+				tuple("Accept-Language", "fr"));
+	}
+
+	@Test
+	public void makeBidsShouldReturnErrorWhenResponseBodyCouldNotBeParsed() {
+		// given
+		final HttpCall<BidRequest> httpCall = givenHttpCall(null, "invalid");
+
+		// when
+		final Result<List<BidderBid>> result = emxDigitalBidder.makeBids(httpCall, null);
+
+		// then
+		assertThat(result.getErrors()).hasSize(1);
+		assertThat(result.getErrors().get(0).getMessage())
+			.startsWith("Failed to decode: Unrecognized token");
+		assertThat(result.getErrors().get(0).getType())
+			.isEqualTo(BidderError.Type.bad_server_response);
+		assertThat(result.getValue()).isEmpty();
+	}
+
+	@Test
+	public void makeBidsShouldReturnEmptyListWhenBidResponseIsNull()
+		throws JsonProcessingException {
+		// given
+		final HttpCall<BidRequest> httpCall = givenHttpCall(null,
+			mapper.writeValueAsString(null));
+
+		// when
+		final Result<List<BidderBid>> result = emxDigitalBidder.makeBids(httpCall, null);
+
+		// then
+		assertThat(result.getErrors()).isEmpty();
+		assertThat(result.getValue()).isEmpty();
+	}
+
+	@Test
+	public void makeBidsShouldReturnEmptyListWhenBidResponseSeatBidIsNull()
+		throws JsonProcessingException {
+		// given
+		final HttpCall<BidRequest> httpCall = givenHttpCall(null,
+			mapper.writeValueAsString(BidResponse.builder().build()));
+
+		// when
+		final Result<List<BidderBid>> result = emxDigitalBidder.makeBids(httpCall, null);
+
+		// then
+		assertThat(result.getErrors()).isEmpty();
+		assertThat(result.getValue()).isEmpty();
+	}
+
+	@Test
+	public void makeBidsShouldReturnBannerBidWhenBannerIsPresentWithChangedBidImpId() throws JsonProcessingException {
```

Comment:
Please change to `makeBidsShouldAlwaysReturnBannerBidWithChangedBidImpId` since they don't have ant matching logic, but always set `banner` type explicitly.